src/main/java/ai/chat/repository/UserRepository.java

```diff
@@ -3,7 +3,9 @@
 import ai.chat.model.User;
 import org.springframework.data.mongodb.repository.MongoRepository;
 
+import java.util.Optional;
+
 public interface UserRepository extends MongoRepository<User, String> {
-  User findByUsername(String username);
+  User findByUserId(String id);
```

Comment:
This should be doublechecked, but the method name should be findById for the repository to give the right implementation.